### DISCUSSION TITLE
How to get theme colors from within a component

### DISCUSSION THREAD

**Discussion by Seanmclem** [mantinedev/mantine#4629] [Q&A]
### Link to the page where something nasty is located

https://mantine.dev/theming/colors/

### Exact quote of what is wrong

A hundred different ways to set theme colors, and very very few specifics on how to consume them.. I have a component I want to use a theme color on, but it's not like a _button text color_  like one of the only examples in the docs for consuming theme colors. How do I get the colors? there's no info about using a hook to do it, or anything. Please and thank you

### Are you willing to create pull request with the fix?

None

**Comment by rtivital** [comment]
To get your theme in any component, use `useMantineTheme` hook – https://mantine.dev/theming/theme-object/#use-mantine-theme-hook. You also get theme object as an argument in `createStyles`, `styles` and `sx` props.

**Reply by Seanmclem** [reply]
Are there examples of that second point? Getting the theme object in like an SX prop? Do you need to use a function?

**Reply by rtivital** [reply]
https://mantine.dev/styles/sx/#subscribe-to-theme

**Reply by Seanmclem** [reply]
Thanks, and am I being ridiculous, or should the subscribe-to-theme stuff maybe be mentioned or linked-to in the Theme documentation, in addition to SX docs? Maybe linked in the Theme>Colors page somewhere?
Same question for the useMantineTheme hook. Shouldn't hooks like that be listed under the "Mantine Hooks" docs?
